### PR TITLE
Feature/preview media fix

### PR DESCRIPTION
**Pull request by MHove89**
Media was not pushed to preview data-source when a data-source was present for preview data. This has been fixed for V7, V8, V9+

### CODE REVIEW

File: src/Enterspeed.Source.UmbracoCms/Composers/EnterspeedComposer.cs [hunk 1]
```diff
@@ -150,7 +151,11 @@ public void Compose(IUmbracoBuilder builder)
 
                 // Media 
                 .Append<EnterspeedMediaPublishJobHandler>()
-                .Append<EnterspeedMediaTrashedJobHandler>();
+                .Append<EnterspeedMediaTrashedJobHandler>()
+
+                // Preview medis
```
Comment (jesperweber) [inline src/Enterspeed.Source.UmbracoCms/Composers/EnterspeedComposer.cs:156]: Typo in _Preview medis_

File: src/Enterspeed.Source.UmbracoCms/Handlers/PreviewMedia/EnterspeedPreviewMediaTrashedJobHandler.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                _enterspeedConnectionProvider.GetConnection(ConnectionType.Preview) != null
+                && job.EntityType == EnterspeedJobEntityType.Media
+                && job.JobType == EnterspeedJobType.Delete
+                && job.ContentState == EnterspeedContentState.Preview;
+        }
+
+        public void Handle(EnterspeedJob job)
+        {
+            var parsed = int.TryParse(job.EntityId, out var parsedId);
+            var media = parsed ? _mediaService.GetById(parsedId) : null;
```
Comment (jesperweber) [inline src/Enterspeed.Source.UmbracoCms/Handlers/PreviewMedia/EnterspeedPreviewMediaTrashedJobHandler.cs:40]: Why do we load the umbraco media. We only use the media.id but we already have the id from the job?

Also _enterspeedIngestService.Delete(media?.Id.ToString(), .....) could result in us sending null as the id parameter how will the enterspeedIngestService.Delete method handle that?

**Review by jesperweber — COMMENTED** [review]
Working and looking good. Only have one comment (and a small typo)

File: src/Enterspeed.Source.UmbracoCms/Handlers/PreviewMedia/EnterspeedPreviewMediaTrashedJobHandler.cs (hunk 2 above)
Comment (MHove89) [reply src/Enterspeed.Source.UmbracoCms/Handlers/PreviewMedia/EnterspeedPreviewMediaTrashedJobHandler.cs:40]: You are right. This seems odd. This is something the has been living for a while in the default way we handled media (Not only for preview). We should probably change that for both the default way we handle media, as well as the preview functionality. What do you think?

File: src/Enterspeed.Source.UmbracoCms/Composers/EnterspeedComposer.cs (hunk 1 above)
Comment (MHove89) [reply src/Enterspeed.Source.UmbracoCms/Composers/EnterspeedComposer.cs]: This has been fixed now :)

File: src/Enterspeed.Source.UmbracoCms/Handlers/PreviewMedia/EnterspeedPreviewMediaTrashedJobHandler.cs (hunk 2 above)
Comment (MHove89) [reply src/Enterspeed.Source.UmbracoCms/Handlers/PreviewMedia/EnterspeedPreviewMediaTrashedJobHandler.cs]: I have fixed it for all job handlers that handles delete now :)

File: src/Enterspeed.Source.UmbracoCms/Handlers/Media/EnterspeedMediaTrashedJobHandler.cs [hunk 3]
```diff
@@ -38,12 +37,18 @@ public bool CanHandle(EnterspeedJob job)
...
         {
             var parsed = int.TryParse(job.EntityId, out var parsedId);
-            var media = parsed ? _mediaService.GetById(parsedId) : null;
 
-            var deleteResponse = _enterspeedIngestService.Delete(media?.Id.ToString(), _enterspeedConnectionProvider.GetConnection(ConnectionType.Publish));
-            if (!deleteResponse.Success && deleteResponse.Status != HttpStatusCode.NotFound)
+            if (parsed)
             {
-                throw new JobHandlingException($"Failed deleting entity ({job.EntityId}/{job.Culture}). Message: {deleteResponse.Message}");
+                var deleteResponse = _enterspeedIngestService.Delete(media?.Id.ToString(), _enterspeedConnectionProvider.GetConnection(ConnectionType.Publish));
```
Comment (jesperweber) [inline src/Enterspeed.Source.UmbracoCms/Handlers/Media/EnterspeedMediaTrashedJobHandler.cs:43]: Should be parseId

**Review by jesperweber — COMMENTED** [review]
Nice, this will also give a small performance improvement. Besides the build error it's looking good.

File: src/Enterspeed.Source.UmbracoCms/Handlers/Media/EnterspeedMediaTrashedJobHandler.cs (hunk 3 above)
Comment (MHove89) [reply src/Enterspeed.Source.UmbracoCms/Handlers/Media/EnterspeedMediaTrashedJobHandler.cs]: Forgot to push :D

**Review by jesperweber — APPROVED** [review]
Awesome

**Pull request merged**